src/main/java/ch/uzh/ifi/hase/soprafs22/controller/ChatController.java

```diff
@@ -64,19 +70,29 @@ public ResponseEntity<List<ChatOverViewGetDTO>> getFirstMessageOfEveryChat(@Requ
 
   @GetMapping("/users/{userId}/chats/{chatId}")
   @ResponseBody
-  public ResponseEntity<ChatMessageGetDTO> getMessagesFromChat(@RequestHeader(value = "authorization", required = false) String token,
+  public ResponseEntity<List<ChatMessageGetDTO>> getMessagesFromChat(@RequestHeader(value = "authorization", required = false) String token,
                                                                @PathVariable(value = "userId") long userId,
                                                                @PathVariable(value = "chatId") long chatId,
                                                                @RequestParam(value = "from", required = false) long from,
                                                                @RequestParam(value = "to", required = false) long to) {
     userService.checkSpecificAccess(token, userId); // 404, 409
 
-    // you can use the mapper here
-    ChatMessageGetDTO chatMessageGetDTO = DTOMapper.INSTANCE.convertMessageToChatMessageGetDTO(new Message());
-    return new ResponseEntity<>(null, null, null);
-  }
 
+    List<Message> messageFromChat = chatService.getMessagesFromChat(chatId, from, to);
 
+    // return the black cards
+      List<ChatMessageGetDTO> chatMessageGetDTOList= new ArrayList<>();
+
+      // mapp the messages
+      for (Message message : messageFromChat){
+          chatMessageGetDTOList.add(DTOMapper.INSTANCE.convertMessageToChatMessageGetDTO(message));
+      }
+
+
+      return new ResponseEntity<>(chatMessageGetDTOList, null, HttpStatus.OK);
+  }
+
+    // TODO: How to implement?
```

Comment:
The reason behind this endpoint is to have a lightweight method call that can be called frequently in the client (so we don't have to make a websocket). For the implementation, you can iterate through the latest messages of every chat of that user, and just return !(message.isRead() && message.getFromUserId() != userId) similar to:
![image](https://user-images.githubusercontent.com/75019468/167309927-b6e06a2f-3e8b-419d-81a0-f10a46c7a44a.png).
(Recommendation, Must)